### PR TITLE
Lagom: Add two-stage build for Fuzzers to enable fuzzing generated code

### LINKED ISSUE #9466: CMake: LibUnicode generated code is not fuzzed
We require the UnicodeDataGenerator code generator in order to generate UnicodeData[.cpp,.h].

When Lagom is built for fuzzing in oss-fuzz, we pass ENABLE_OSS_FUZZ=ON, which causes all executable targets to link against either libfuzzer or the AFL equivalent.

https://github.com/google/oss-fuzz/blob/5e5d08420a0fbe6613d8bcf94a1f8ce15b66b525/projects/serenity/build.sh#L19-L30

This causes a main() function to be pre-defined that expects `LLVMFuzzerTestOneInput` as the entry point to our code.

Trying to build our code generator tools in this mode is a build error.

One possible design might be to do a two-stage build in the oss-fuzz repo. The first stage is without fuzzing, and generates and installs the code generators (with BUILD_LAGOM=OFF). If we modify the unicode_data.cmake file, or Lagom/CMakeLists.txt to allow setting a custom IMPORTED target for the UnicodeDataGenerator when fuzzing, we could fuzz the generated code as well.

Doing this will be much easier after https://github.com/SerenityOS/serenity/pull/9281 is merged.

### PR DESCRIPTION
**Pull request by ADKaster**
Fixes #9466

The script could of course use some elbow grease but this build gets the basics down.

Merging this will break oss-fuzz until the [companion merge request](https://github.com/google/oss-fuzz/pull/7309) in the oss-fuzz repository is merged.

### CODE REVIEW

**Review by linusg — CHANGES_REQUESTED** [review]
Typos in first commit message: `LibJs` -> `LibJS`, `Fuzzili` -> `Fuzzilli`

File: Userland/Libraries/LibAudio/Buffer.h [hunk 1]
```diff
@@ -115,6 +113,14 @@ class Buffer : public RefCounted<Buffer> {
     {
     }
 
+    // Empty Buffer representation, to avoid tiny anonymous buffers in EOF states
+    Buffer()
+        : m_buffer()
+        , m_id(-1)
+        , m_sample_count(0)
```
Comment (trflynn89) [inline Userland/Libraries/LibAudio/Buffer.h:120]: This seems like the plain types should be initialized at the declaration:
```c++
    const i32 m_id { -1 };
    const int m_sample_count { 0 };
```

Then this ctor can just be:
```c++
Buffer() = default;
```

**Comment by trflynn89** [comment]
LGTM otherwise :)

File: Userland/Libraries/LibAudio/Buffer.h (hunk 1 above)
Comment (ADKaster) [reply Userland/Libraries/LibAudio/Buffer.h]: oh, yep, that's a cleaner solution. Using = default now.

**Comment by ADKaster** [comment]
Changes since last push:

- Fixed Tim's comment about the Buffer commit
- Fixed typos in fuzzer commit body (looks like my dictionary needs some help 😅 )
- Added updated docs to the first commit.

**Review by linusg — APPROVED** [review]

**Pull request merged**